File: src/main/java/com/smart/appsa/model/enums/StatusPedido.java
package com.smart.appsa.model.enums;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum StatusPedido {
    PENDENTE(1),
    PRODUCAO(2),
    CONCLUIDO(3);

    Integer value;

    @JsonCreator
    public static StatusPedido fromValue(String value) {
        Integer parsed = Integer.valueOf(value);
        for (StatusPedido s : values()) {
            if (parsed.equals(s.value)) return s;
        }
        throw new IllegalArgumentException("StatusPedido inválido: " + value);
    }

    private StatusPedido(Integer value) {
        this.value = value;
    }

    public Integer getValue() { return value; }
}

File: frontend/planta-smart-4.0/src/domain/enums/StatusPedido.ts
export enum StatusPedido {
  Pendente,
  Producao,
  Concluido,
}

export const StatusPedidoToInt: Record<StatusPedido, number> = {
  [StatusPedido.Pendente]: 1,
  [StatusPedido.Producao]: 2,
  [StatusPedido.Concluido]: 3,
};

export const StatusPedidoStringToEnum: Record<string, StatusPedido> = {
  PENDENTE: StatusPedido.Pendente,
  PRODUCAO: StatusPedido.Producao,
  CONCLUIDO: StatusPedido.Concluido,
};
